core/modules/parsers/wikiparser/rules/html.js

```diff
@@ -60,6 +64,27 @@ exports.parse = function() {
 			var reEnd = new RegExp("(" + reEndString + ")","mg");
 			tag.children = this.parser.parseInlineRun(reEnd,{eatTerminator: true});
 		}
+		tag.end = this.parser.pos;
+		tag.closeTagEnd = tag.end;
+		if (tag.closeTagEnd === tag.openTagEnd || this.parser.source[tag.closeTagEnd - 1] !== '>') {
+			tag.closeTagStart = tag.end;
+		} else {
+			tag.closeTagStart = tag.closeTagEnd - 2;
+			var closeTagMinPos = tag.children.length > 0 ? tag.children[tag.children.length-1].end : tag.openTagEnd;
+			if (!Number.isSafeInteger(closeTagMinPos)) closeTagMinPos = tag.openTagEnd;
+			while (tag.closeTagStart >= closeTagMinPos) {
+				var char = this.parser.source[tag.closeTagStart];
+				if (char === '>') {
+					tag.closeTagStart = -1;
+					break;
+				}
+				if (char === '<') break;
+				tag.closeTagStart -= 1;
+			}
+			if (tag.closeTagStart < closeTagMinPos) {
+				tag.closeTagStart = tag.end;
+			}
+		}
```

Comment:
@Gk0Wk @Jermolene we should make sure that this PR does not introduce any significant performance regression.

Reply:
I think this won't cost much, because it just `while` from `>` to the `<` in `<div>`, won't step more than `<blockquote>`

But why would we need open/closeTagStart/End ? Can this feature be turn on when needed?